### ISSUE TITLE
Panic: runtime error - 2.0.29 

### ISSUE THREAD

**Issue by 5ch17** [DNSCrypt/dnscrypt-proxy#998]
2.0.29 on FreeBSD crashes with "Panic: runtime" error.

**Comment by jedisct1** [comment]
Can you post your `dnscrypt-proxy.toml` configuration somewhere?

**Comment by 5ch17** [comment]
https://cryptpad.fr/code/#/2/code/edit/t3JAEJtbmlGv749ZHW0okjGj

**Comment by egypcio** [comment]
@5ch17 , @jedisct1 ,
I also confirm this panic. it affected also **FreeBSD** 12 and HEAD (13). that 'linux' mentioned over there might be related to any cross-compiling magics, which we do not have here compiling `dnscrypt-proxy` in our ports tree - but it crashes anyway.

good news is, so far, I can definitely say that checking ceed905 out it works just fine. from 220d418 on, it breaks. the panic is generated right after querying the daemon (I did use the default config to start it, and just ran `drill` to query a hostname's addr).

```
root@localhost:~ # drill -t -p 5353 @127.0.0.1 netbsd.org
Error: error sending query: General LDNS error
```

**Comment by welwood08** [comment]
Looking at the changes in 220d418, I imagine the cause of the panic is that `app.Start()` is now never called in the case where `svc == nil` (my editor can find no references to it being called at all), which means that `proxy.InitPluginsGlobals()` is never called, which means the pointers to the various plugin slices are not initialised by the time a query arrives and tries to use plugins.

**Comment by welwood08** [comment]
After testing on my FreeBSD 10.3 based system, I can confirm the regression commit identified by @egypcio and that 37c9394 fixes it.

**Comment by egypcio** [comment]
I do confirm it works for **FreeBSD** _11.3, 12.0 and 13.0_: I did test it all myself. Thanks!
A new port version (2.0.29_2) with this new commit should land in a few.

**Comment by jedisct1** [comment]
Awesome! Good job everybody! 🎉 

Since this probably affects more than FreeBSD, I'm going to release version 2.0.30 soon.

**Comment by namtsui** [comment]
The proposed fix works on OpenBSD. I experienced a similar issue with broken DNS when trying to update to 2.0.29. Thank you.

**Comment by 5ch17** [comment]
Confirmed that fix + 2.0.30 release/binary works on FreeBSD 11.2, thanks for quick turnaround.
